File: frontend/src/core/geo/entities/geo-region.ts
export interface GeoRegion {
  id: string;
  countryCode: string;
  level: 'country' | 'province' | 'municipality';
  name: string;
  latitude?: number;
  longitude?: number;
  active: boolean;
}

export interface ProvinceOption {
  id: string;
  name: string;
}

export interface MunicipalityOption {
  id: string;
  name: string;
}

File: backend/inventory-app/src/main/java/com/inventory/domain/model/geo/GeoRegion.java
package com.inventory.domain.model.geo;

import java.math.BigDecimal;
import java.util.UUID;

public class GeoRegion {
    private final UUID id;
    private final String countryCode;
    private final String level;
    private final String name;
    private final UUID parentId;
    private final BigDecimal latitude;
    private final BigDecimal longitude;
    private final boolean active;

    public GeoRegion(UUID id, String countryCode, String level, String name,
                     UUID parentId, BigDecimal latitude, BigDecimal longitude, boolean active) {
        this.id = id != null ? id : UUID.randomUUID();
        this.countryCode = countryCode;
        this.level = level;
        this.name = name;
        this.parentId = parentId;
        this.latitude = latitude;
        this.longitude = longitude;
        this.active = active;
    }

    public UUID getId() { return id; }
    public String getCountryCode() { return countryCode; }
    public String getLevel() { return level; }
    public String getName() { return name; }
    public UUID getParentId() { return parentId; }
    public BigDecimal getLatitude() { return latitude; }
    public BigDecimal getLongitude() { return longitude; }
    public boolean isActive() { return active; }
}
